syntax = "proto3";

option java_multiple_files = true;
option java_package = "com.stockearte.tp1_grupo10.grpc";

import "tienda.proto";
import "google/protobuf/empty.proto";

// Definición del mensaje Stock
message Stock {
  int64 id = 1;
  Tienda tienda = 2;
  string producto_codigo = 3;
  int32 cantidad = 4;
}

// Definición de mensajes para las operaciones del servicio
message StockIdRequest {
  int64 id = 1;
}

message StockList {
  repeated Stock stocks = 1;
}

message StockUpdateRequest {
  int64 id = 1;
  int32 cantidad = 2;
}

service StockService {
  rpc add (Stock) returns (Stock);
  rpc getOneById (StockIdRequest) returns (Stock);
  rpc getAll (google.protobuf.Empty) returns (StockList);
  rpc update (Stock) returns (Stock);
  rpc updateStockQuantity (StockUpdateRequest) returns (Stock);
  rpc delete (StockIdRequest) returns (google.protobuf.Empty);
}